### ISSUE TITLE
Downloading data from the CN network

### ISSUE THREAD

**Issue by ElodieK** [obspy/obspy#2000]
I am trying to use the mass downloader to retrieve seismograms from various networks. I have located an error with the network CN. There seems to be a problem with the xml files. A few mseed and xml files download and then these errors appear:

KeyError Traceback (most recent call last) /Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/util/deprecation_helpers.py in getattr(self, name) 60 try: ---> 61 real_module_name = self.__import_map[name] 62 except: KeyError: 'core' During handling of the above exception, another exception occurred: AttributeError Traceback (most recent call last) /Users/elodiekendall/Desktop/0PATHS/OBSPY/globalcomb.py in () 105 106 sta_xml_file_list = glob.glob(os.path.join("stations/*.xml")) --> 107 inv = read_inventory(sta_xml_file_list[0]) 108 for sta_xml in sta_xml_file_list[1:]: 109 inv += read_inventory(sta_xml) in read_inventory(path_or_file_object, format) /Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/util/decorator.py in _map_example_filename(func, *args, **kwargs) 296 except IOError: 297 pass --> 298 return func(*args, **kwargs) 299 return _map_example_filename 300 /Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/inventory/inventory.py in read_inventory(path_or_file_object, format) 70 return read_inventory(fh.name, format=format) 71 return _read_from_plugin("inventory", path_or_file_object, ---> 72 format=format)[0] 73 74 /Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/util/base.py in _read_from_plugin(plugin_type, filename, format, **kwargs) 460 raise TypeError(msg % (format_ep.name, ', '.join(eps))) 461 # read --> 462 list_obj = read_format(filename, **kwargs) 463 return list_obj, format_ep.name 464 /Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/io/stationxml/core.py in _read_stationxml(path_or_file_object) 153 networks = [] 154 for network in root.findall(_ns("Network")): --> 155 networks.append(_read_network(network, _ns)) 156 157 inv = obspy.core.inventory.Inventory(networks=networks, source=source, /Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/io/stationxml/core.py in _read_network(net_element, _ns) 191 192 def _read_network(net_element, _ns): --> 193 network = obspy.core.inventory.Network(net_element.get("code")) 194 _read_base_node(net_element, network, _ns) 195 network.total_number_of_stations = \ /Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/util/deprecation_helpers.py in getattr(self, name) 61 real_module_name = self.__import_map[name] 62 except: ---> 63 raise AttributeError 64 msg = ("Module '%s' is deprecated and will stop working with the " 65 "next ObsPy version. Please import module '%s'instead.") % ( AttributeError: In [36]:

The script ran fine for other networks such as II or IU.
I'm hoping you can help so that I don't have to download this network's data separately.

Thanks,
Elodie

**Comment by krischer** [comment]
Hi,

a couple things before we can try to help you:

(1) Can you paste the error log in a form that preserves line breaks?
(2) Are you running the latest ObsPy version 1.1.0?
(3) Can you paste an example script to reproduce the errror?

Cheers!

**Comment by ElodieK** [comment]
Hi,


  1.  Yes sorry about that.


KeyError                                  Traceback (most recent call last)

/Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/util/deprecation_helpers.py in __getattr__(self, name)

     60         try:

---> 61             real_module_name = self.__import_map[name]

     62         except:


KeyError: 'core'


During handling of the above exception, another exception occurred:


AttributeError                            Traceback (most recent call last)

/Users/elodiekendall/Desktop/0PATHS/OBSPY/globalcomb.py in <module>()

    105

    106 sta_xml_file_list = glob.glob(os.path.join("stations/*.xml"))

--> 107 inv = read_inventory(sta_xml_file_list[0])

    108 for sta_xml in sta_xml_file_list[1:]:

    109    inv += read_inventory(sta_xml)


<decorator-gen-195> in read_inventory(path_or_file_object, format)


/Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/util/decorator.py in _map_example_filename(func, *args, **kwargs)

    296                         except IOError:

    297                             pass

--> 298         return func(*args, **kwargs)

    299     return _map_example_filename

    300


/Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/inventory/inventory.py in read_inventory(path_or_file_object, format)

     70             return read_inventory(fh.name, format=format)

     71     return _read_from_plugin("inventory", path_or_file_object,

---> 72                              format=format)[0]

     73

     74


/Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/util/base.py in _read_from_plugin(plugin_type, filename, format, **kwargs)

    460         raise TypeError(msg % (format_ep.name, ', '.join(eps)))

    461     # read

--> 462     list_obj = read_format(filename, **kwargs)

    463     return list_obj, format_ep.name

    464


/Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/io/stationxml/core.py in _read_stationxml(path_or_file_object)

    153     networks = []

    154     for network in root.findall(_ns("Network")):

--> 155         networks.append(_read_network(network, _ns))

    156

    157     inv = obspy.core.inventory.Inventory(networks=networks, source=source,


/Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/io/stationxml/core.py in _read_network(net_element, _ns)

    191

    192 def _read_network(net_element, _ns):

--> 193     network = obspy.core.inventory.Network(net_element.get("code"))

    194     _read_base_node(net_element, network, _ns)

    195     network.total_number_of_stations = \


/Users/elodiekendall/anaconda3/lib/python3.5/site-packages/obspy/core/util/deprecation_helpers.py in __getattr__(self, name)

     61             real_module_name = self.__import_map[name]

     62         except:

---> 63             raise AttributeError

     64         msg = ("Module '%s' is deprecated and will stop working with the "

     65                "next ObsPy version. Please import module '%s'instead.") % (


AttributeError:


2. I haven't changed to the latest version yet but I was hoping to process the rest of my data first.

3. Yes here is my script, the path obviously needs changing.


#!/usr/bin/env python
import obspy
import rotate_fdsn
import obspy.io.sac
from obspy import read, read_inventory
from obspy.core import UTCDateTime
import glob
import os
from obspy.geodetics import gps2dist_azimuth
#functions for trace station lat,long
def getstationcoordinates(inventory,stream):
     for tr in stream:
       for netw in inventory.networks:
         for sta in netw.stations:
           if tr.stats.station == sta.code and netw.code == tr.stats.network:
             for cha in sta.channels:
               if tr.stats.location == cha.location_code:
                 tr.stats.latitude = cha.latitude
                 tr.stats.longitude = cha.longitude
     return
path = '/Users/elodiekendall/Desktop/0PATHS/OBSPY/GLOBAL/100/'
pathsyn = '/Users/elodiekendall/Desktop/0PATHS/OBSPY/GLOBAL/100/syn/'

# #----------------- fold: t and origintime/lat/lon (3)
# os.mkdir(path+'syn/')
# model_list = ['ANIS', 'ISO', 'S40', 'SAVANI']
# #loop over model_list
# for model in model_list:

# st = obspy.read(os.path.join(path,model,'*.sac'))

# ## 3 interp and cut
# st.interpolate(sampling_rate=1)

# ## 4 rm, rt, tap
# period_list = ['0', '25', '40', '60', '100']
# st.detrend(type='demean')
# st.detrend(type='linear')
# st.taper(0.05,type='hann')
# tapered_st = st.copy()

# ## 5 bp
# for period in period_list:
# #deepcopy
# st = tapered_st.copy()
# if period == '0':
# st.filter('bandpass',freqmin=1/200,freqmax=1/20,corners=4, zerophase=True)
# elif period  == '25':
# st.filter('bandpass',freqmin=1/(25*1.1),freqmax=1/(25*0.9),corners=4, zerophase=True)
# elif period == '40':
# st.filter('bandpass',freqmin=1/(40*1.2),freqmax=1/(40*0.8),corners=4, zerophase=True)
# elif period == '60':
# st.filter('bandpass',freqmin=1/(60*1.2),freqmax=1/(60*0.8),corners=4, zerophase=True)
# elif period == '100':
# st.filter('bandpass',freqmin=1/(100*1.25),freqmax=1/(100*0.75),corners=4, zerophase=True)

# ## 3 interp and cut
# for tr in st:
# t = tr.stats.starttime +1.21125
# tr.trim(t,t+5000)
# ## 6 Write out as sac format='SAC')
# tr.write(os.path.join(pathsyn,tr.stats.channel[2]+'.'+model+'.'+tr.stats.station+'.SAC.P.F'+period), format='SAC')
# print(t)

# -----------------
#Download data
os.chdir(path)
from obspy.clients.fdsn.mass_downloader import CircularDomain, \
    Restrictions, MassDownloader

    ####### OPTIONSS
t=obspy.UTCDateTime(2011,8,10,23,45,43.000000)
origin_time = obspy.UTCDateTime(2011, 8, 10, 23, 45, 43)
lat=-7.1029
lon=-12.8059

minrad=20
maxrad=140
aval=6378137.0
fval=0.0033528106647474805
# Circular domain around the epicenter. This will download all data between
# 70 and 90 degrees distance from the epicenter.
domain = CircularDomain(latitude=lat, longitude=lon,
                        minradius=minrad, maxradius=maxrad)
restrictions = Restrictions(
#careful with spaces!
network='CN',
# network='AK,CN,PM,AI,ZM,AU,YT,AF,XB,JP,NZ,ND,XP,PS',
# network='II,IU,G,AK,GE,NO,DK,CN,IC,PM,AI,ZM,AU,YT,AF,XB,JP,NZ,ND,XP,BK,PS,'NY',
    starttime=origin_time,
    endtime=origin_time + 6000,
    reject_channels_with_gaps=True,
    minimum_length=0.95,
    minimum_interstation_distance_in_m=10E3,
    channel_priorities=("LH[Z12]","BH[Z12]", "HH[Z12]","LH[ZNE]", "BH[ZNE]","HH[ZNE]"),
    # location_priorities=["00"], sanitize=True)
location_priorities=["00", ""])

# No specified providers will result in all known ones being queried.
mdl = MassDownloader()
# The data will be downloaded to the ``./waveforms/`` and ``./stations/``
mdl.download(domain, restrictions, mseed_storage="real",
             stationxml_storage="stations")

sta_xml_file_list = glob.glob(os.path.join("stations/*.xml"))
inv = read_inventory(sta_xml_file_list[0])
for sta_xml in sta_xml_file_list[1:]:
   inv += read_inventory(sta_xml)

st = obspy.read(os.path.join("real/*.mseed"))


Thanks a lot,

Elodie

________________________________

**Comment by krischer** [comment]
I can run this

```python
from obspy.clients.fdsn.mass_downloader import CircularDomain, \
    Restrictions, MassDownloader

import obspy

t=obspy.UTCDateTime(2011,8,10,23,45,43.000000)
origin_time = obspy.UTCDateTime(2011, 8, 10, 23, 45, 43)
lat=-7.1029
lon=-12.8059

minrad=20
maxrad=140
aval=6378137.0
fval=0.0033528106647474805
# Circular domain around the epicenter. This will download all data between
# 70 and 90 degrees distance from the epicenter.
domain = CircularDomain(latitude=lat, longitude=lon,
                        minradius=minrad, maxradius=maxrad)
restrictions = Restrictions(
#careful with spaces!
network='CN',
# network='AK,CN,PM,AI,ZM,AU,YT,AF,XB,JP,NZ,ND,XP,PS',
# network='II,IU,G,AK,GE,NO,DK,CN,IC,PM,AI,ZM,AU,YT,AF,XB,JP,NZ,ND,XP,BK,PS,'NY',
    starttime=origin_time,
    endtime=origin_time + 6000,
    reject_channels_with_gaps=True,
    minimum_length=0.95,
    minimum_interstation_distance_in_m=10E3,
    channel_priorities=("LH[Z12]","BH[Z12]", "HH[Z12]","LH[ZNE]", "BH[ZNE]","HH[ZNE]"),
    # location_priorities=["00"], sanitize=True)
location_priorities=["00", ""])

# No specified providers will result in all known ones being queried.
mdl = MassDownloader()
# The data will be downloaded to the ``./waveforms/`` and ``./stations/``
mdl.download(domain, restrictions, mseed_storage="real",
             stationxml_storage="stations")
```

just fine. Please update to the latest version and see if it fixes your problem. We don't have the manpower to support outdated ObsPy versions.

**Comment by ElodieK** [comment]
Hi again,


I have updated obspy:

# Anaconda Python Distribution
# `conda install -c obspy obspy` is no longer supported for most platforms!
$ conda update -c conda-forge obspy


Now,



________________________________

**Comment by ElodieK** [comment]
**


Now the error:

/Users/elodiekendall/anaconda3/lib/python3.5/site-packages/scipy/linalg/basic.py:1226: RuntimeWarning: internal gelsd driver lwork query error, required iwork dimension not returned. This is likely the result of LAPACK bug 0038, fixed in LAPACK 3.2.2 (released July 21, 2010). Falling back to 'gelss' driver.

  warnings.warn(mesg, RuntimeWarning)


The mseed nor xml files have downloaded.


I see in github someone has written: This is likely harmless and safe to ignore. However the mass download has not worked so it's not harmless.


Thanks,

Elodie

________________________________

**Comment by krischer** [comment]
This is a warning from scipy if you have a really old lapack. It just states that it has to fall back to an older (and maybe less efficient) implementation. If I had to guess I'd assume that your scipy is compiled against apple's accelerate framework which tends to be really old. If you create a new conda environment it should either install numpy + scipy linked against the MKL or openblas (I think) for conda-forge libraries.

Nonetheless - this is only a warning and given that everything downloaded I assume that ObsPy worked correctly. I'm confused about your last statement to be honest - if it worked it is indeed harmless for your particular installation, no?
